File: NDPServer/src/main/java/com/slender/config/manager/FilterConfigManager.java
package com.slender.config.manager;

import com.slender.exception.config.URLConfigNotFoundException;
import jakarta.servlet.Filter;

import java.util.HashMap;

public final class FilterConfigManager {
    private final static HashMap<Class<? extends Filter>,String> filterConfigs=new HashMap<>();

    public FilterConfigManager addFilterConfig(Class<? extends Filter> filter, String url){
        filterConfigs.put(filter, url);
        return this;
    }

    public String getFilterURL(Class<? extends Filter> filter){
        String url = filterConfigs.getOrDefault(filter, null);
        if(url==null) throw new URLConfigNotFoundException();
        return url;
    }
}

File: NDPServer/src/main/java/com/slender/config/manager/FilterConfigManager.kt
package com.slender.config.manager

import com.slender.exception.config.URLConfigNotFoundException
import jakarta.servlet.Filter

class FilterConfigManager {
    private val filterConfigs = HashMap<Class<out Filter>, String>()
    fun addFilterConfig(filter: Class<out Filter>, url: String): FilterConfigManager {
        filterConfigs[filter] = url
        return this
    }

    fun getFilterURL(filter: Class<out Filter>): String
        = filterConfigs.getOrDefault(filter, null) ?: throw URLConfigNotFoundException()
}
